### DISCUSSION TITLE
LED driver failure

### DISCUSSION THREAD

**Discussion by rywc** [pyPhotometry/code#18] [Q&A]
We have purchased two photometry acquisition boards so far, one of which seems to have a problem with the LED1 driver. We checked and found that it's not attributed to the detector or LED itself, as the issue occurs only on analog 1 with the '2EX_1EM_pulsed' mode, and the problem is solved if the acquisition board is swapped. The Oeps Support Team suggested I format PYBFLASH and reset the micropython files, but it didn't solve the problem. Do you have any suggestions for this issue?

![image](https://github.com/user-attachments/assets/756d41ea-eaee-4f47-bb8d-6db271f02189)

**Comment by ThomasAkam** [comment]
Hi @rywc, can you provide a bit more information about the nature of the problem: Does the LED not light up at all? How does the image in your previous post relate to the problem? What do the signals look like if you put the system in continuous illumination mode, set the LED 2 current to 0 and then vary the LED 1 current?

**Reply by rywc** [reply]
Hi @ThomasAkam, thanks for your reply. The LED light lights up, but the signal fluctuates, and I assumed this is because the LED driver is unstable. The picture in my previous post shows the voltage output from the detector without connecting a sample with 2EX_1EM_pulsed mode; ie, both analog 1&2 signals were supposed to be stable, but only analog1 was noisy. 
![Screenshot 2024-08-05 135646](https://github.com/user-attachments/assets/85d6f1da-58c1-4218-9b1a-77b4cd26746f)

When I operate in continuous illumination mode with LED2 current set to 0, analog1 input fluctuates, as I saw in 2EX_1EM mode.
![Screenshot 2024-08-05 140612](https://github.com/user-attachments/assets/c0d242dd-43cf-4926-809f-bfe54d3d3b71)

**Reply by ThomasAkam** [reply]
Hmmm, I've not seen this issue before. Could you try swapping which LED is plugged into which LED driver output and then acquiring signal in 2EX_1EM_pulsed mode.  If it is the led driver that is causing the problem then channel 1 will still fluctuate (even though it is driving the led that was previously fine) and channel 2 will be steady (driving the led that was previously fluctuating)

**Comment by rywc** [comment]
Here's the result of swapping the LEDs plugged into LED1 and LED2. Again, the analog1 input fluctuates, consistent with swapping the acquisition board solved the problem, which indicates it's not attributed to the LEDs or the detector.
![Screenshot 2024-08-08 143207](https://github.com/user-attachments/assets/79e18adc-1cd6-4de2-855d-39018e03e043)

**Reply by ThomasAkam** [reply]
I agree this looks like a problem with the board, either with the LED driver circuit, or with the analog output of the pyboard that is used to control the LED current.
